python/ray/data/_internal/planner/plan_udf_map_op.py

```diff
@@ -606,3 +608,194 @@ def _create_map_transformer_for_block_based_map_op(
         BuildOutputBlocksMapTransformFn.for_blocks(),
     ]
     return MapTransformer(transform_fns, init_fn=init_fn)
+
+
+_SENTINEL = object()
+
+T = TypeVar("T")
+U = TypeVar("U")
+
+
+def _generate_transform_fn_for_async_map(
```

Comment:
Moved this down to make review hopefully a bit easier -- this essentially is a new implementation